test/types/community/collection/insertX.test-d.ts

```diff
@@ -123,12 +123,14 @@ await collectionWithObjectId.insertOne({
   numberField: 23,
   fruitTags: ['hi']
 });
-// should not demand _id if it is ObjectId
-await collectionWithObjectId.insertOne({
-  stringField: 'hola',
-  numberField: 23,
-  fruitTags: ['hi']
-});
+// if _id is defined on the schema, it must be passed to insert operations
+expectError(
+  collectionWithObjectId.insertOne({
+    stringField: 'hola',
+    numberField: 23,
+    fruitTags: ['hi']
+  })
+);
```

Comment:
can we add some extra cases below to make sure we aren't allowed to insert an objectId in place of a user defined 1) required numeric _id 2) optional numeric _id to cover the manifestations of the bugs that we're fixing here?